### PR TITLE
8289612: Change hotspot/jtreg tests to not use Thread.stop

### PR DESCRIPTION
**Pull request by lmesnik**
The tests are updated to don't use Thread.stop(). Tests whose intention is to verify async exception updated to use jvmti StopThread.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8289612](https://bugs.openjdk.org/browse/JDK-8289612): Change hotspot/jtreg tests to not use Thread.stop

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [628ff010](https://git.openjdk.org/jdk/pull/9505/files/628ff01085eafe6ca6bfa8f5eeb9041fcad62bd3)
 * [Daniel D. Daugherty](https://openjdk.org/census#dcubed) (@dcubed-ojdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9505/head:pull/9505` \
`$ git checkout pull/9505`

Update a local copy of the PR: \
`$ git checkout pull/9505` \
`$ git pull https://git.openjdk.org/jdk pull/9505/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9505`

View PR using the GUI difftool: \
`$ git pr show -t 9505`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9505.diff">https://git.openjdk.org/jdk/pull/9505.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/gc/gctests/mallocWithGC2/mallocWithGC2.java [hunk 1]
```diff
@@ -113,8 +113,6 @@ public void run() {
                                 tArray[i].start();
 
                         tArray[0].join(); // wait for the javaHeapEater Thread to finish
-                        tArray[1].stop(); // Once javaHeapEater is finished, stop the
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/vmTestbase/gc/gctests/mallocWithGC2/mallocWithGC2.java:116]: So without this the other thread will run for a full 3 minutes - is that a concern?

**Review by dholmes-ora — COMMENTED** [review]
Hi Leonid,

Can we avoid all the agent stuff by defining a JVMTI helper class in the test library:

public class JVMTI {
  private native void stopThread(Thread t, Throwable ex);
  public void stopThread(Thread t) {
    stopThread(t, new ThreadDeath());
  }
}

And in cpp file just use GetEnv to get JVMTIEnv and call stopThread?

Otherwise the conversions seem quite reasonable.

Thanks.

File: test/hotspot/jtreg/runtime/Thread/AsyncExceptionTest.java [hunk 2]
```diff
@@ -48,14 +48,16 @@ public class AsyncExceptionTest extends Thread {
...
     private boolean receivedThreadDeathinInternal2 = false;
 
+    public static native int stopThread(Thread thread);
+
     @Override
     public void run() {
         try {
             internalRun1();
         } catch (ThreadDeath td) {
             throw new RuntimeException("Catched ThreadDeath in run() instead of internalRun2() or internalRun1(). receivedThreadDeathinInternal1=" + receivedThreadDeathinInternal1 + "; receivedThreadDeathinInternal2=" + receivedThreadDeathinInternal2);
```
Comment (AlanBateman) [inline test/hotspot/jtreg/runtime/Thread/AsyncExceptionTest.java:58]: Drive-by comment: the exception messages mean the lines are 240+ characters line and make it impossible to see changes when using side-by-side diffs. Maybe someday it should be trimming down to something sane.

File: test/hotspot/jtreg/vmTestbase/gc/gctests/mallocWithGC2/mallocWithGC2.java [hunk 3]
```diff
@@ -113,8 +113,6 @@ public void run() {
...
-                        // the cHeapEater thread.
                 } catch (Exception e) {
                         throw new TestFailure("Test Failed.", e);
                 }
```
Comment (AlanBateman) [inline test/hotspot/jtreg/vmTestbase/gc/gctests/mallocWithGC2/mallocWithGC2.java:118]: Drive-by comment on this source file is that it looks like it uses 8-space indent everywhere, maybe tabs were converted to 8 spaces by mistake?

**Comment by AlanBateman** [comment]
Thanks for doing this. I agree with David that AsyncExceptionOnMonitorEnter AsyncExceptionTest doesn't really need to be started with -agentlib. The can_signal_thread capability can be obtained in the alive phase so JNI code can obtain a JVMTI environment and add this capability.

File: test/hotspot/jtreg/vmTestbase/nsk/stress/stack/stack002.java [hunk 4]
```diff
@@ -147,7 +152,7 @@ public void run() {
              return;
              };
              ***/
-            tester.stop();
+            tester.stop = true;
```
Comment (AlanBateman) [inline test/hotspot/jtreg/vmTestbase/nsk/stress/stack/stack002.java:155]: Can the comment "The test hangs on JDK 1.2.2 Classic VM" be removed?

**Review by dcubed-ojdk — CHANGES_REQUESTED** [review]
Just a few comments/observations.

File: test/hotspot/jtreg/vmTestbase/gc/gctests/mallocWithGC2/mallocWithGC2.java (hunk 1 above)
Comment (lmesnik) [reply test/hotspot/jtreg/vmTestbase/gc/gctests/mallocWithGC2/mallocWithGC2.java:116]: VM doesn't wait for the completion of threads that are executed native code. Seems like a new bug?

File: test/hotspot/jtreg/vmTestbase/gc/gctests/mallocWithGC2/mallocWithGC2.java (hunk 3 above)
Comment (lmesnik) [reply test/hotspot/jtreg/vmTestbase/gc/gctests/mallocWithGC2/mallocWithGC2.java:118]: there are a lot of old tests with the 8-spaces tab. I don't want to fix it in this bugfix. Makes sense?

**Review by lmesnik — COMMENTED** [review]
Fixed stopThread using and other comments (or replied to them).

File: test/hotspot/jtreg/runtime/Thread/AsyncExceptionTest.java [hunk 5]
```diff
@@ -53,13 +54,17 @@ public void run() {
         try {
             internalRun1();
         } catch (ThreadDeath td) {
-            throw new RuntimeException("Catched ThreadDeath in run() instead of internalRun2() or internalRun1(). receivedThreadDeathinInternal1=" + receivedThreadDeathinInternal1 + "; receivedThreadDeathinInternal2=" + receivedThreadDeathinInternal2);
+            throw new RuntimeException("Catched ThreadDeath in run() instead of internalRun2() or internalRun1().\n"
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/Thread/AsyncExceptionTest.java:57]: existing: s/Catched/Caught/

File: test/hotspot/jtreg/runtime/Thread/AsyncExceptionTest.java [hunk 6]
```diff
@@ -53,13 +54,17 @@ public void run() {
...
+                    + "receivedThreadDeathinInternal1=" + receivedThreadDeathinInternal1
+                    + "; receivedThreadDeathinInternal2=" + receivedThreadDeathinInternal2);
         } catch (NoClassDefFoundError ncdfe) {
-            // ignore because we're testing Thread.stop() which can cause it
+            // ignore because we're testing StopThread() which can cause it
         }
 
         if (receivedThreadDeathinInternal2 == false && receivedThreadDeathinInternal1 == false) {
-            throw new RuntimeException("Didn't catched ThreadDeath in internalRun2() nor in internalRun1(). receivedThreadDeathinInternal1=" + receivedThreadDeathinInternal1 + "; receivedThreadDeathinInternal2=" + receivedThreadDeathinInternal2);
+            throw new RuntimeException("Didn't catched ThreadDeath in internalRun2() nor in internalRun1().\n"
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/Thread/AsyncExceptionTest.java:65]: Existing: s/catched/catch/

File: test/hotspot/jtreg/vmTestbase/gc/gctests/mallocWithGC2/mallocWithGC2.java (hunk 1 above)
Comment (dholmes-ora) [reply test/hotspot/jtreg/vmTestbase/gc/gctests/mallocWithGC2/mallocWithGC2.java:116]: @lmesnik I don't understand what you mean here. The cHeapEater thread is a non-daemon thread so the test program won't terminate until it does. In the old code we would assist it to terminate early using stop() but in the new code it will run until it completes normally after 180 seconds.

**Comment by lmesnik** [comment]
Yep, the goal of this thread is just to call malloc/free in the cycle. Doesn't need to wait for completion.
